File: v3/typescript/search-all-helper.ts
import { AxiosRequestConfig } from "axios";
import { NewscatcherError } from "./error";

export async function searchAll(method: Function, requestParameters: any, options?: AxiosRequestConfig) {
    try {
        const pageSize = requestParameters.page_size || requestParameters.pageSize || 100;
        const response = (await method(requestParameters, options)).data;

        if (response.total_hits > pageSize) {
            for (let page = 2; page <= response.total_pages; page++) {
                requestParameters.page = page;
                const res = (await method(requestParameters, options)).data;
                response.articles.push(...res.articles);
            }
        }
        return response.articles;
    } catch (e) {
        if (e instanceof NewscatcherError) 
            console.error("Exception when calling API:", e);
        else 
            console.error("Unexpected error:", e);
        return null;
    }
}

File: v3/python/newscatcherapi_client/search_all_helper.py
import newscatcherapi_client.exceptions

def search_all(method, **kwargs):
    try:
        if "page_size" in kwargs:
            page_size = kwargs["page_size"]
        elif "pageSize" in kwargs:
            page_size = kwargs["pageSize"]
        else:
            page_size = 100

        response = method(**kwargs)

        if response.total_hits > page_size:
            for page in range(2, response.total_pages + 1):
                kwargs["page"] = page
                res = method(**kwargs)
                response.articles.extend(res.articles)

        return response.articles

    except newscatcherapi_client.exceptions.ApiException as e:
        print("Exception when calling Search API: %s\n" % e)
        return None